src/Vts.Scripting/ForwardSolvers/FS01_ROfRhoAndFtSingle.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Vts.Common;
+using Vts.Modeling.ForwardSolvers;
+using Plotly.NET.CSharp;
+
+namespace Vts.Scripting.ForwardSolvers;
+
+/// <summary>
+/// Class using the Vts.dll library to demonstrate predicting reflectance as a function of rho and time frequency
+/// </summary>
+public class FS01_ROfRhoAndFtSingle : IDemoScript
```

Comment:
Now that these are classes, I think we need to revisit the names, these do not conform to any naming convention. I understand the need for the numbering to match the MATLAB numbers but we have to come up with a better solution.

Reply:
Strongly disagree on this specific comment - these need to be script-first and match all the target destination files as the most helpful, most discoverable and browsable by Biophotonics  engineers (not developers). They're only classes to add testability, but never to be consumed directly by anyone else. Suggest we make them only visible to the test library as a compromise (`InternalsVisibleTo` attribute).

Reply:
I said revisit, at least let us discuss? What about Example01ROfRhoAndFtSingle - I would like to follow Pascal case and avoid underscores. The FS is redundant since they are in a folder named ForwardSolvers.

Yes I agree, these would be better as internal.

Reply:
I think we are discussing. :) If the rationale for my position wasn't clear, I am happy to elaborate, but I am also fine with my opinion not being the winning argument. The proposed rationale provided as feedback for changing naming was because they were now classes. I don't think that's the best rationale in this specific case, which is why I disagreed. Here's a little more context for my stated opinion (which again, doesn't have to win, as it's ultimately the decision of the project lead):

These particular scripts existing in the Vts solution serve a specific, narrow purpose - to make testable code that would otherwise not be tested/compiled alongside the library that they reference, because they're destined to be in notebooks of various flavors. They are not intended to provide callable code from other parts of the codebase.

Therefore, I was proposing that we reconsider the "standard" C# naming conventions with respect to the consumer of this code ("the customer"), which is someone who will be opening a notebook, and running the examples with a single mouse click. While we do not (yet) have the final shape of what those example files will look like, my initial thinking was that they would all be in the same folder (right now in Matlab, they're all shoved in 2 files), which would mean that their names _would_ in fact need to disambiguate from each other across the different prefixes. The workflow of this user, in my view would be:

```pwsh
git clone https://github.com/VirtualPhotonics/VTS.git
cd VTS
python3 FS01_ + [TAB] to complete the name, FS01_ROfRhoAndFtSingle
```
This is what I meant by:

> the most helpful, most discoverable and browsable by Biophotonics engineers (not developers)

The scripts currently exist in folders only because I thought it would make them easier to refactor if our library dependencies change (e.g. if the solver demos in the future no longer needed to pull in a segregated "Vts.MonteCarlo.dll"). I could also see moving them all into a single "Examples" folder. In either case, I think a) imposing standard C# idioms onto code that is not intended to be called from outside C# consumers degrades the experience of the above use case without any obvious benefit (to me, at least - I'm open to feedback...), and b) that the underscores makes them more readable. In other words, I think it makes good sense to "work backwards" - to envision the final destination/goal, and design these names to match/meet that goal. (But if they don't match because they have to be different, ok).

My hope is that the names here can be shared with their intended destination to minimize any disconnect (e.g. `ForwardSolverExample01ROfRhoAndFtSinglecs` becoming `FS01_ROfRhoAndFtSingle.ipynb` seems confusing to me, and the latter is much more readable IMO).

Thoughts on any of the above?

Reply:
Think I need to update the user story above, to disambiguate a .py script vs a .ipynb notebook. Command line use might not be as important as I was originally thinking. Will revise the stories tomorrow.

Reply:
That all makes sense. Do the names for Python have to be FS01_ etc? Do you think that someone using the Python scripts would be looking at the C# ones to need to make the correlation?

I think matching the MATLAB names as much as possible is good because those have been around for a while.

I think a good way around this would be to have Vts.Scripting as a separate repo also, then it is not putting code that does not conform to naming in the same project as Vts. This will also help the tests because we can leave them as is.

Do you think repo names like:
Vts.Scripting.CSharp
Vts.Scripting.Python 
would be good?

Reply:
User stories are helpful :) I think I've talked myself around...

The likely workflow of the .NET Polyglot Notebook or Jupyter user would be:

```cmd
git clone https://github.com/VirtualPhotonics/VTS.git
cd VTS
code .
```

The last line opens the folder for visual browsing within VS Code. They'd then interact with the examples directly. If it's primarily a visual organization system, then it would be best to break things up into folders. 

Would `ForwardSovlers\Demo01ROfRhoAndFtSingle` and similar work?

> I think a good way around this would be to have Vts.Scripting as a separate repo also

`Vts.Scripting` was by design intentionally in the main repo so it could be compiled alongside the main code as developers do their work, and tested as part of the CI. Then, the plan was to create separate repos with notebooks (that used Nuget to pull in `Vts.dll`), which would be _mostly_ copy-pastable into .ipynb notebooks. I was thinking the naming might be something along the lines of :

vts-demos-python-notebooks
vts-demos-matlab
vts-demos-dotnet-notebooks

How's that sound? Long living "base" prefix `vts-demos` and then we can grow over time. Say, we want "raw" Python and JavaScript demos too, then we can add:

vts-demos-python
vts-demos-javascript
vts-demos-rust
vts-demos-go
vts-demos-swift
...

That work for you?

Reply:
Also, I've now updated the 28 functions to make plotly chart rendering optional, and now skip it in the Vts.Scripting.Test tests.  Most viable alternative to mocking Plotly, which would require building undesired layers of indirection and hamper the intended purpose, which is to expose directly the full plotly capabilities/options to the script consumers. I spent a couple hours on mocking alternatives, and it's just not feasible based on the way Plotly.NET works, and what our intention is for users. I can explain more in-person if anyone's interested.

Reply:
`Would ForwardSovlers\Demo01ROfRhoAndFtSingle and similar work?`

Yes! I like that and if we name it correctly then it's ok to stay in the main repo. 

I'm good with the naming for the other repos because hyphens are the standard for those languages.

I think we went through a similar process with unit testing for Plotly and came to same conclusion so I like that we can skip the plotting in the tests.